pkg/phlaredb/block_querier.go

```diff
@@ -983,17 +986,29 @@ func (b *singleBlockQuerier) SelectMatchingProfiles(ctx context.Context, params
 			lbls = make(phlaremodel.Labels, 0, 6)
 		}
 	}
-	pIt := query.NewJoinIterator(
-		0,
-		[]query.Iterator{
+
+	var (
+		buf       [][]parquet.Value
+		joinIters []query.Iterator
+	)
+
+	if b.meta.Version >= 2 {
+		joinIters = []query.Iterator{
 			b.profiles.columnIter(ctx, "SeriesIndex", newMapPredicate(lblsPerRef), "SeriesIndex"),
 			b.profiles.columnIter(ctx, "TimeNanos", query.NewIntBetweenPredicate(model.Time(params.Start).UnixNano(), model.Time(params.End).UnixNano()), "TimeNanos"),
 			b.profiles.columnIter(ctx, "StacktracePartition", nil, "StacktracePartition"),
```

Comment:
Problem #1: This requires the column to be there, maybe should have a orNull predicate or something.